torch/fft/__init__.py

```diff
@@ -9,18 +9,233 @@
+
+    >>> torch.fft.irfft(T)
+    tensor([-0.4000,  0.8392,  1.1298,  2.0702,  2.3608])
+
+    So, it is recommended to always pass the signal length :attr:`n`:
+
+    >>> torch.fft.irfft(T, t.numel())
+    tensor([0., 1., 2., 3.])
+""")
+
+hfft = _add_docstr(_fft.fft_hfft, r"""
+hfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the one dimensional discrete Fourier transform of a hermitian
+symmetric :attr:`input` signal.
+
+Because the signal is hermitian in the time-domain, the result will be real in
+the frequency domain. Note that some input frequencies must be purely real to
+satisfy the hermitian property. In these cases the imaginary component will be
+ignored.
+
+Args:
+    input (Tensor): the input tensor representing a half-hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        real output. If given, the input will either be zero-padded or trimmed to this
+        length before taking the FFT. Defaults to odd output: ``n=input.size(dim) * 2 - 1``.
+    dim (int, optional): The dimension along which to take the one dimensional FFT.
+    norm (str, optional): Normalization mode:
+        ``"forward"`` - the forward transform (:func:`~torch.fft.hfft`) is normalized by ``1/n``,
+        ``"backward"`` - the backward transform (:func:`~torch.fft.ihfft`) is normalized by ``1/n``,
+        ``"ortho"`` - both transform directions are normalized by ``1/sqrt(n)`` (making it orthonormal).
+        Default is ``"backward"``.
+
+Example::
+
+    Taking a frequency signal that's purely real and bringing into the time
+    domain gives hermitian symmetric output:
+
+    >>> import torch.fft
+    >>> t = torch.arange(4)
+    >>> t
+    tensor([0, 1, 2, 3])
+    >>> T = torch.fft.ifft(t)
+    >>> T
+    tensor([ 1.5000+-0.0000j, -0.5000-0.5000j, -0.5000+-0.0000j, -0.5000+0.5000j])
+
+    Note that ``T[1] == T[-1].conj()`` is redundant. We can thus compute the
+    forward transform without considering negative frequencies:
+
+    >>> torch.fft.hfft(T[:3], n=4)
+    tensor([0., 1., 2., 3.])
+
+    Like with :func:`torch.fft.irfft`, the output length must be given in order
+    to recover an even length output:
+
+    >>> torch.fft.hfft(T[:3])
+    tensor([-1.5000,  1.6521,  0.2432,  2.5410,  1.4590,  3.7568,  2.3479])
+""")
+
+ihfft = _add_docstr(_fft.fft_ihfft, r"""
+ihfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the inverse one dimensional Fourier transform of a real signal.
+
+The IFFT of a real signal is hermitian-symmetric, ``X[i] = conj(X[-i])`` so
+the output contains only the positive frequencies below the nyquist frequency.
+To compute the full output, use :func:`torch.fft.ifft`.
+
+Args:
+    input (Tensor): the real input tensor
+    n (int, optional): FFT length. If given, the input will either be zero-padded
+        or trimmed to this length before taking the FFT.
+    dim (int, optional): The dimension along which to take the one dimensional IFFT.
```

Comment:
IFFT or IHFFT?

Reply:
Changed all of these to either real (I)FFT or Hermitian (I)FFT.